modules/database/database.go

```diff
@@ -0,0 +1,153 @@
+package database
+
+import (
+	"database/sql"
+	"fmt"
+	"testing"
+
+	// Microsoft SQL Database Driver
+	_ "github.com/denisenkom/go-mssqldb"
+
+	// PostgreSQL Database Driver
+	_ "github.com/lib/pq"
+
+	// MySQL Database Driver
+	_ "github.com/go-sql-driver/mysql"
```

Comment:
That sounds a bit odd. Typically, if you checkout a project and run `dep ensure`, all it should do is download the dependencies for that project into the `vendor` folder. You can then run `dep ensure -add github.com/go-sql-driver/mysql` to add a dependency, which will modify both `Gopkg` files.

Reply:
My workflow is, first directing to database directory and then running `dep ensure`. After that, all dependencies including `gruntwork-io/terratest` are downloaded into vendor. `Gopkg.lock` becomes really big but 'Gopkg.toml' does not change. If I would like to execute `dep ensure -add`, it tells me that the dependency is already imported or required, so -add is only valid with a constraint, which means that adding dependencies fails. I wonder whether I'm doing something incorrectly.

Reply:
Hi @foreverXZC, you should run `dep ensure` in the root of `terratest` (where the existing `Gopkg.toml` and `Gopkg.lock` files are). `dep` will find all the dependencies below that and will make sure they are installed for the project.

Reply:
Hi @mcalhoun , after I run `dep ensure` in the root of terratest, `Gopkg.lock` becomes long and contains terratest packages. I think that's because test examples requires these things, so `dep ensure` install them into `Gopkg.lock`. My suggestion is that I only keep `modules` folder and then run `dep ensure`, so terratest packages will not be included in `Gopkg.lock`. What's your opinion?

Reply:
Hi @foreverXZC...I believe the reason you are seeing the terratest packages included in `Gopkg.lock` is because they are imported with fully qualified references to our repos ("github.com/gruntwork-io/terratest/...") and your repo is `foreverXZC/terratest` so dep sees those as external dependencies. This is normal behavior.

I will have to do a bit of research on the best way for us to merge your PR (any PR that includes dependency changes). I guess we will have to run `dep ensure` on the merged master branch (which will remove the external dependencies on `gruntwork-io/terratest`) and then commit the `Gopkg.lock` back to the master branch.

The tl;dr is run `dep ensure` on from the root of your repo, do work commit it to your PR and we'll deal with it when we merge (unless @brikis98 or someone else has a better idea of how to handle this).